lib/hanami/model/configurator.rb

```diff
@@ -25,12 +25,22 @@ class Configurator
       # @api private
       attr_reader :_schema
 
+      # @since x.x.x
+      # @api private
+      attr_reader :callbacks
+
       # @since 0.7.0
       # @api private
       def self.build(&block)
         new.tap { |config| config.instance_eval(&block) }
       end
 
+      # @since x.x.x
+      # @api private
+      def initialize
+        @callbacks = Hash[before: [], after: []]
```

Comment:
well...if only i had a crazy idea:-)
I like the `gateway` idea you presented. Not sure if `gateway` has limitations for the future? But if I understood the comment from @solnic there will always be a `gateway` but not necessarily a `connection` so 'gateway'
 is even better.